### DISCUSSION TITLE
P3016 Can I send the CASCADE hint to prisma migrate reset?

### DISCUSSION THREAD

**Discussion by loren** [prisma/prisma#15369] [Q&A] [answered]
When I try to reset, I get this error:
```
% npx prisma migrate reset --force
Error: P3016

The fallback method for database resets failed, meaning Migrate could not clean up the database entirely. Original error:
db error: ERROR: cannot drop table "Foo" because other objects depend on it
DETAIL: policy foobar_policy_all on table "FooBar" depends on table "Foo"
HINT: Use DROP ... CASCADE to drop the dependent objects too.
   0: sql_migration_connector::best_effort_reset
             at migration-engine/connectors/sql-migration-connector/src/lib.rs:290
   1: migration_core::state::Reset
             at migration-engine/core/src/state.rs:341
```

I'm happy to drop the dependent objects as I'm trying to reset everything. Is there a workaround?

**Comment by austincrim** [comment, marked as answer]
Hey @loren 👋,

It's not possible to pass that kind of flag to `migrate reset`, though that sounds like a good idea for a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) if you don't mind filling one out.

In the meantime, you could possibly use `prisma migrate diff` to get to the equivalent SQL. Run something like this in your terminal:
```bash
npx prisma migrate diff --from-schema-datasource=path/to/schema.prisma --to-empty --script
```

Then add the necessary `CASCADE` keywords to the outputted SQL.

**Reply by fafrd** [reply]
Feature request: https://github.com/prisma/prisma/issues/15422
